lib/hanami/utils/kernel.rb

```diff
@@ -404,15 +413,14 @@ def self.Integer(arg)
       #   # Missing #respond_to?
       #   input = BasicObject.new
       #   Hanami::Utils::Kernel.BigDecimal(input) # => TypeError
-      def self.BigDecimal(arg)
+      def self.BigDecimal(arg, precision: BIGDECIMAL_DEFAULT_PRECISION)
         case arg
+        when ->(a) { a.respond_to?(:to_d) && a.class == Rational } then arg.to_d(precision)
```

Comment:
@jodosha in JRuby Rational needs an argument, that argument is the precision, the other object responds to_d method but the don't need the precision.